chore(deps): pin dependencies

Octobob opened the pull request:
This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [node](https://togithub.com/nodejs/node) | engines | pin | [`^14.0.0` -> `14.21.1`](https://renovatebot.com/diffs/npm/node/v14.21.1/14.21.1) |
| [pnpm](https://pnpm.io) ([source](https://togithub.com/pnpm/pnpm)) | engines | pin | [`^7.9.5` -> `7.15.0`](https://renovatebot.com/diffs/npm/pnpm/7.15.0/7.15.0) |

📌 **Important**: Renovate will wait until you have merged this Pin PR before creating any *upgrade* PRs for the affected packages. Add the preset `:preserveSemverRanges` to your config if you instead don't wish to pin dependencies.

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR is behind base branch, or you tick the rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get [config help](https://togithub.com/renovatebot/renovate/discussions) if that's undesired.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [Renovate Bot](https://togithub.com/renovatebot/renovate).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OctopusDeploy/step-package-template?pullRequest=99) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OctopusDeploy/step-package-template?pullRequest=99) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OctopusDeploy/step-package-template?pullRequest=99) it.</sub>

Octobob commented:
### ⚠ Artifact update problem

Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

 - any of the package files in this branch needs updating, or 
 - the branch becomes conflicted, or
 - you click the rebase/retry checkbox if found above, or
 - you rename this PR's title to start with "rebase!" to trigger it manually

The artifact failure details are included below:

##### File name: pnpm-lock.yaml

```
installing v2 tool node v14.21.3
updating node-gyp
+ node-gyp@9.4.0
added 115 packages from 49 contributors and updated 1 package in 6.252s
linking tool node v14.21.3
node: v14.21.3 /usr/local/bin/node
npm: 6.14.18  /usr/local/bin/npm
Installed v2 /usr/local/buildpack/tools/v2/node.sh in 9 seconds
installing v2 tool pnpm v7.9.5
+ pnpm@7.9.5
added 1 package in 0.991s
linking tool pnpm v7.9.5
7.9.5
Installed v2 /usr/local/buildpack/tools/v2/pnpm.sh in 2 seconds
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your pnpm version is incompatible with "/tmp/renovate/repos/github/OctopusDeploy/step-package-template".

Expected version: 7.33.4
Got: 7.9.5

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".

```
